python/doc/examples/calibration/bayesian_calibration/plot_hierarchical_calibration_fission_gas.py

```diff
@@ -0,0 +1,577 @@
+ot.ResourceMap.SetAsScalar("Distribution-QMax", 1.0)
+
+# %%
+# Create the list of all samplers in the Gibbs algorithm as outlined in the chart below,
+# where direct samplers are represented in green and random walk Metropolis-Hastings samplers in blue.
+#
+# .. figure:: ../../_static/fission_gas_mh.png
+#     :align: center
+#     :alt: Metropolis-Hastings-within-Gibbs description
+#     :width: 100%
+#
+# We start with the samplers of :math:`\mu_{\mathrm{diff}}, \mu_{\mathrm{crack}}`.
+# We are able to directly sample these conditional distributions,
+# hence we use the :class:`~openturns.RandomVectorMetropolisHastings` class.
+
+samplers = [
+    ot.RandomVectorMetropolisHastings(
+        mu_rv,
+        initial_state,
+        [i],
+        ot.Function(PosteriorParametersMu(dim=i, lb=lbs[i], ub=ubs[i])),
+    )
+    for i in range(ndim)
+]
+
+# %%
+# We continue with the samplers of :math:`\sigma_{\mathrm{diff}}^2, \sigma_{\mathrm{crack}}^2`.
+# We are alse able to directly sample these conditional distributions.
+
+samplers += [
+    ot.RandomVectorMetropolisHastings(
+        sigma_square_rv,
+        initial_state,
+        [ndim + i],
+        ot.Function(
+            PosteriorParametersSigmaSquare(
+                dim=i, lb=lbs_sigma_square[i], ub=ubs_sigma_square[i]
+            )
+        ),
+    )
+    for i in range(ndim)
+]
+
+# %%
+# We finish with the samplers of the :math:`\vect{x}_i`, with :math:`1 \leq i \leq n_{exp}`.
+# Each of these samplers outputs points in a :math:`\sampleSize_{\mathrm{exp}}`-dimensional space.
+# We are not able to directly sample these conditional posterior distributions,
+# so we resort to random walk Metropolis-Hastings.
+
+for exp in range(nexp):
+    base_index = 2 * ndim + ndim * exp
+
+    samplers += [
+        ot.RandomWalkMetropolisHastings(
+            ot.Function(PosteriorLogDensityX(exp=exp)),
+            support,
+            initial_state,
+            ot.Normal([0.0] * 2, [6.0, 0.15]),
+            [base_index + i for i in range(ndim)],
+        )
+    ]
+
+# %%
+# The Gibbs algorithm combines all these samplers.
+
+sampler = ot.Gibbs(samplers)
+x_desc = []
+for i in range(1, nexp + 1):
+    x_desc += [f"x_{{{label}, {i}}}" for label in desc]
+sampler.setDescription(mu_desc + sigma_square_desc + x_desc)
+
+# %%
+# Run this Metropolis-within-Gibbs algorithm and check the acceptance rates
+# for the Random walk Metropolis-Hastings samplers.
+
+samples = sampler.getSample(2000)
+acceptance = [
+    sampler.getMetropolisHastingsCollection()[i].getAcceptanceRate()
+    for i in range(6, len(samplers))
+]
+
+adaptation_factor = [
+    sampler.getMetropolisHastingsCollection()[i]
+    .getImplementation()
+    .getAdaptationFactor()
+    for i in range(6, len(samplers))
+]
+
+print("Minimum acceptance rate = ", np.min(acceptance))
+print("Maximum acceptance rate for random walk MH = ", np.max(acceptance[2 * ndim :]))
+
+# %%
+# Plot the posterior distribution
```

Comment:
Following the discussion on the "MCMC diagnostic" maybe it is necessary to point out that the obtained samples should be post-treated in order to obtained representative samples of the posterior ?
The MCMC diagnostic is not available yet in OpenTURNS but it is important in the example to mention this point for non familiar MCMC user.